smithy-trait-codegen/src/main/java/software/amazon/smithy/traitcodegen/integrations/javadoc/SmithyGeneratedInterceptor.java

```diff
@@ -5,29 +5,22 @@
 
 package software.amazon.smithy.traitcodegen.integrations.javadoc;
 
-import software.amazon.smithy.model.traits.DeprecatedTrait;
 import software.amazon.smithy.traitcodegen.sections.ClassSection;
 import software.amazon.smithy.traitcodegen.writer.TraitCodegenWriter;
 import software.amazon.smithy.utils.CodeInterceptor;
+import software.amazon.smithy.utils.SmithyGenerated;
 
 /**
- * Adds the {@code @Deprecated} annotation to generated classes if the smithy shape the class corresponds to
- * has the {@link DeprecatedTrait} trait applied.
+ * Adds the {@link software.amazon.smithy.utils.SmithyGenerated} annotation to all generated classes.
  */
-final class DeprecatedAnnotationClassInterceptor implements CodeInterceptor.Prepender<ClassSection,
-        TraitCodegenWriter> {
-    @Override
-    public void prepend(TraitCodegenWriter writer, ClassSection section) {
-        writer.writeWithNoFormatting("@Deprecated");
-    }
-
+public class SmithyGeneratedInterceptor implements CodeInterceptor.Prepender<ClassSection, TraitCodegenWriter> {
     @Override
     public Class<ClassSection> sectionType() {
         return ClassSection.class;
     }
 
     @Override
-    public boolean isIntercepted(ClassSection section) {
-        return section.shape().hasTrait(DeprecatedTrait.class);
+    public void prepend(TraitCodegenWriter writer, ClassSection section) {
+        writer.write("@$T", SmithyGenerated.class);
```

Comment:
Why doesn't this get applied up where the other annotations are applied? The strategy seems inconsistent.